### PR TITLE
Bug FIx: PCI logging

### PR DESCRIPTION
**Pull request by jinhoyrivos**
- arrayvec uses sdt (not checked by #![cfg_attr(not(test), no_std)])
- re-implemented logging wihout arrayvec

Bug=[SW-8085]

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -4,9 +4,6 @@
 
 #![cfg_attr(not(test), no_std)]
 
-use arrayvec::ArrayString;
```
Comment (furquan-rivos) [inline src/lib.rs]: Does it not work to disable the std features in arrayvec i.e. make it no_std compatible?

File: Cargo.toml [hunk 2]
```diff
@@ -10,7 +10,7 @@ edition = "2021"
 [dependencies]
 # implicitly defines a feature like "dep:log"
 log = { version = "0.4.14", default-features = false, optional = true }
-arrayvec = "*"
```
Comment (furquan-rivos) [inline Cargo.toml:13]: Commit message needs update.

File: src/lib.rs [hunk 3]
```diff
@@ -1074,7 +1074,7 @@ impl Device {
     }
 
     fn print_config_space_bytes(&self, num_bytes: usize) {
-        const BUFFER_SIZE: usize = 100;
+        const BUFFER_SIZE: usize = 70;
```
Comment (furquan-rivos) [inline src/lib.rs:1077]: unrelated change?

Comment (jinhoyrivos) [reply src/lib.rs:1077]: I didn't see this reviews, sorry. I updated the commit msg and made separate commits for the changed buffer sizes
